### PR TITLE
File saving, electron build, tree navigation, settings page, platform integration via token

### PR DESCRIPTION
**Pull request by ivanbatic**
This change is [Reviewable](https://reviewable.io/reviews/rabix/cottontail-frontend/22)

### CODE REVIEW

File: config/webpack.test.js [hunk 1]
```diff
@@ -132,15 +132,15 @@ module.exports = {
 
             /**
              * Raw loader support for *.css files
-             * Returns file content as string
+             * Returns file tabData as string
```
Comment (mayacoda) [inline config/webpack.test.js]: Overzealous search and replace?

File: src/app/components/code-editor/code-editor.component.ts [hunk 2]
```diff
@@ -1,82 +1,78 @@
...
+        //         this.eventHub.publish(new UpdateFileAction(file));
+        //     })
+        // );
+        //
+        // this.subs.push(
+        //     this.editor.validationResult
+        //         .subscribe((result: ValidationResponse) => {
+        //             this.eventHub.publish(new CwlValidationResult(result));
+        //         })
+        // );
```
Comment (mayacoda) [inline src/app/components/code-editor/code-editor.component.ts]: Validation has been moved up to tabManager, can we remove this commented code?

File: src/app/components/code-editor/code-editor.ts [hunk 3]
```diff
@@ -1,76 +1,48 @@
...
 
-        this.contentChanges = Observable.fromEvent(this.editor as any, "change")
-            .debounceTime(300)
-            .map(_ => this.document.getValue())
-            .distinctUntilChanged()
-            .withLatestFrom(this.fileStream, (content, file) => {
-                return Object.assign(file, {content});
-            }).share();
+        this.contentType.subscribe(type => {
+            this.setMode("json");
```
Comment (mayacoda) [inline src/app/components/code-editor/code-editor.ts]: Language should not be hardcoded. For SB apps, we should assume type to be JSON, though.

File: src/app/components/layout/layout.types.ts [hunk 4]
```diff
@@ -27,7 +27,7 @@ export class PanelGroup {
 
         // If the panel doesn't exist, throw an exception, it's probably a mistake in the code
         if (index === -1) {
-            throw `Trying to toggle non-existing panel “${id}”. 
+            throw `Trying to toggle non-existing panel “${id}”. ¡
```
Comment (mayacoda) [inline src/app/components/layout/layout.types.ts]: Typo? :)

File: src/app/components/tab-manager/tab-manager.component.ts [hunk 5]
```diff
@@ -1,71 +1,70 @@
...
+        //         case "Workflow":
+        //             this.type = "workflow";
+        //             break;
+        //         default:
+        //             this.type = "text";
+        //             break;
+        //     }
+        //
+        //     this.isLoading = false;
+        // }));
```
Comment (mayacoda) [inline src/app/components/tab-manager/tab-manager.component.ts]: Remove commented code.

File: src/app/components/tree-view/tree-view.component.scss [hunk 6]
```diff
@@ -80,7 +82,7 @@ ct-tree-view {
 
 .name-container {
   display: inline-block;
-  width: calc(100% - 10px);
+  width: calc(100% - 20px);
```
Comment (mayacoda) [inline src/app/components/tree-view/tree-view.component.scss]: Consider using variables for these kinds of values.

File: src/app/components/workspace/registry/component-registry.ts [hunk 7]
```diff
@@ -35,21 +35,21 @@ export class ComponentRegistry {
     /**
      * Extracts the Components from the GL's component hierarchy.
      *
-     * @param content GL's "content" attribute which contains a tree of rows/columns/components
+     * @param content GL's "tabData" attribute which contains a tree of rows/columns/components
```
Comment (mayacoda) [inline src/app/components/workspace/registry/component-registry.ts]: Again, overzealous search and replace? Also, do we still need the whole workspace component and accompanying services?

File: src/app/components/code-editor/code-editor.ts (hunk 3 above)
Comment (ivanbatic) [inline src/app/components/code-editor/code-editor.ts]: And since SB apps are all we have at the moment...

Comment (mayacoda) [inline src/app/components/code-editor/code-editor.ts]: Not for long, though. :D  And doesn't ` src/app/components/panels/sb-public-apps-panel.component.ts:62` already hardcode the language for SB apps?

**Pull request merged**